### PR TITLE
[android][brightness] Migrate to Expo Modules API

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
Continue migration to Expo Modules API.

# How
Followed the typical migration steps.

# Test Plan
Tested in bare expo. All the system brightness methods currently don't work as they require permission only available to system apps. I suggest deprecating all the system methods. This would mean the permission request could be removed and the API would match what's available on iOS.

### CODE REVIEW

File: packages/expo-brightness/android/src/main/java/expo/modules/brightness/BrightnessModule.kt [hunk 1]
```diff
@@ -1,115 +1,68 @@
...
+        try {
+          val lp = currentActivity.window.attributes
+          lp.screenBrightness = brightnessValue
+          currentActivity.window.attributes = lp // must be done on UI thread
+          promise.resolve(null)
+        } catch (e: Exception) {
+          promise.reject(SetBrightnessException())
+        }
       }
     }
```
Comment (lukmccall) [inline packages/expo-brightness/android/src/main/java/expo/modules/brightness/BrightnessModule.kt:41]:
```suggestion
    AsyncFunction("setBrightnessAsync") { brightnessValue: Float ->
      try {
        val lp = currentActivity.window.attributes
        lp.screenBrightness = brightnessValue
        currentActivity.window.attributes = lp // must be done on UI thread
      } catch (e: Exception) {
        throw SetBrightnessException())
      }    
    }.runOnQueue(Queues.MAIN)
```

File: packages/expo-brightness/android/src/main/java/expo/modules/brightness/BrightnessModule.kt [hunk 2]
```diff
@@ -1,115 +1,68 @@
...
+        val brightness = if (lp.screenBrightness == WindowManager.LayoutParams.BRIGHTNESS_OVERRIDE_NONE) {
+          // system brightness is not overridden by the current activity, so just resolve with it
+          getSystemBrightness()
+        } else {
+          lp.screenBrightness
+        }
+
+        promise.resolve(brightness)
       }
     }
```
Comment (lukmccall) [inline packages/expo-brightness/android/src/main/java/expo/modules/brightness/BrightnessModule.kt:55]:
```suggestion
    AsyncFunction("getBrightnessAsync") { ->
      val lp = currentActivity.window.attributes
      val brightness = if (lp.screenBrightness == WindowManager.LayoutParams.BRIGHTNESS_OVERRIDE_NONE) {
        // system brightness is not overridden by the current activity, so just resolve with it
        getSystemBrightness()
      } else {
        lp.screenBrightness
      }
      
      return@AsyncFunction brightness
    }.runOnQueue(Queues.MAIN)
```

File: packages/expo-brightness/android/src/main/java/expo/modules/brightness/BrightnessModule.kt [hunk 3]
```diff
@@ -122,68 +75,63 @@ class BrightnessModule(
...
-  }
 
-  @ExpoMethod
-  fun restoreSystemBrightnessAsync(promise: Promise) {
-    val activity = currentActivity
-    activity.runOnUiThread {
-      try {
-        val lp = activity.window.attributes
+    AsyncFunction("restoreSystemBrightnessAsync") {
+      currentActivity.runOnUiThread {
```
Comment (lukmccall) [inline packages/expo-brightness/android/src/main/java/expo/modules/brightness/BrightnessModule.kt:81]: Please, use `.runOnQueue(Queues.MAIN)` instead

File: packages/expo-brightness/android/src/main/java/expo/modules/brightness/BrightnessModule.kt [hunk 4]
```diff
@@ -122,68 +75,63 @@ class BrightnessModule(
...
-  }
 
-  @ExpoMethod
-  fun isUsingSystemBrightnessAsync(promise: Promise) {
-    val activity = currentActivity
-    activity.runOnUiThread {
-      val lp = activity.window.attributes
-      promise.resolve(lp.screenBrightness == WindowManager.LayoutParams.BRIGHTNESS_OVERRIDE_NONE)
+    AsyncFunction("isUsingSystemBrightnessAsync") { promise: Promise ->
+      currentActivity.runOnUiThread {
```
Comment (lukmccall) [inline packages/expo-brightness/android/src/main/java/expo/modules/brightness/BrightnessModule.kt:89]: Same here ;)

**Review by lukmccall — CHANGES_REQUESTED** [review]

**Review by lukmccall — APPROVED** [review]

**Review by tsapeta — APPROVED** [review]

**Pull request merged**